searchcore/src/vespa/searchcore/proton/server/executorthreadingservice.h

```diff
@@ -18,12 +18,14 @@ class ExecutorThreadingServiceStats;
 class ExecutorThreadingService : public searchcorespi::index::IThreadingService
 {
 private:
+    vespalib::ThreadStackExecutorBase & _sharedExecutor;
     vespalib::ThreadStackExecutor _masterExecutor;
     vespalib::BlockingThreadStackExecutor _indexExecutor;
     vespalib::BlockingThreadStackExecutor _summaryExecutor;
     ExecutorThreadService _masterService;
     ExecutorThreadService _indexService;
     ExecutorThreadService _summaryService;
+    ExecutorThreadService _sharedService;
```

Comment:
ExecutorThreadService and the IThreadService interface is used to represent a single thread. This should not be used for the shared executor. I think we should just expose the shared executor as a vespalib::ThreadExecutor (or ThreadStackExecutorBase if needed).

Reply:
Done